internal/reorder/stackbranch.go

```diff
@@ -19,11 +21,60 @@ type StackBranchCmd struct {
 	Parent string
 	// The name of the trunk branch.
 	// Mutually exclusive with --parent.
+	// The branch can be rooted at a given commit by appending "@<commit>" to the
+	// branch name.
 	Trunk string
 }
 
 func (b StackBranchCmd) Execute(ctx *Context) error {
-	panic("not implemented")
+	tx := ctx.DB.WriteTx()
+	defer tx.Abort()
+
+	branch, _ := tx.Branch(b.Name)
+	var parentState meta.BranchState
+
+	// Figure out which commit we need to start this branch at.
+	var headCommit string
+
+	if b.Trunk != "" {
+		parentState.Name, headCommit, _ = strings.Cut(b.Trunk, "@")
+		parentState.Trunk = true
+	} else {
+		// We assume the parent branch (if not set manually) is the previous branch
+		// in the reorder operation.
+		if b.Parent == "" {
+			b.Parent = ctx.State.Branch
+		}
+		if b.Parent == "" {
+			return ErrInvalidCmd{
```

Comment:
Hm. I wonder if we can add a special check on this before start running commands. Optional, though.

Reply:
Yeah, it's possible but a little more complex than I wanted to do here. It would require a step between parsing and executing. Maybe we can add it in the future.